pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
> If a single delete call to the API is responsible for the delete and is returning a 404, doesn't that seem like an issue that should be fixed in the API, not client side?

The issue I tried to solve is #19403. I've tried to do that at the api server side, but the race seems inevitable, unless we overhaul the entire grace-period implementation (discussed in https://github.com/kubernetes/kubernetes/issues/20497#issuecomment-190976488), but no way we can to that at 1.2. Then I thought of solving it on the client-side. It seems to me the easiest & safest solution would be "check the existence of the resource, return error if the object doesn't exist, otherwise submits the delete and ignore the NotFound error if it occurs"